indexing.sql

```diff
@@ -0,0 +1,59 @@
+SELECT DISTINCT make_title
+FROM car_models
+WHERE make_code = 'LAM';
+-- 22.135ms
+
+SELECT DISTINCT model_title
+FROM car_models
+WHERE make_code = 'NISSAN' and model_code = 'GT-R';
+-- 13.503ms
+
+SELECT make_code, model_code, model_title, year
+FROM car_models
+WHERE make_code = 'LAM';
+-- 18.601ms
+
+SELECT *
+FROM car_models
+WHERE year BETWEEN 2010 and 2015;
+-- 191.071ms
+
+SELECT *
+FROM car_models
+WHERE year = 2010;
+-- 48.563ms
+
+CREATE INDEX codeIndex
+ON car_models (make_code, model_code);
```

Comment:
You only need an index on `make_code`. This would create a nested index where all the make_codes are sorted, then within in make_code the model_code would be sorted.

Reply:
You can actually keep this index because of the second query, but at least have an index for just `make_code`.